python/cli.rst

```diff
@@ -0,0 +1,68 @@
+##############################
+Python Command Line Interfaces
+##############################
+
+Command line interfaces (CLI) in Python may be implemented using the ``argparse`` module or using `Click`_.
+
+.. _Click: https://click.palletsprojects.com
+
+.. _cli-using-click:
+
+Command Line Interfaces using Click
+===================================
+
+Click is a Python package for creating command line interfaces in a composable way. It supports:
+
+* arbitrary nesting of commands
+* automatic help page generation
+* lazy loading of subcommands at runtime
+
+`Click`_ has good `documentation`_ including a `quickstart`_ guide.
+
+.. _documentation: https://click.palletsprojects.com/en/7.x/#documentation
+.. _quickstart: https://click.palletsprojects.com/en/7.x/quickstart/
```

Comment:
A little unfortunate to hard-code a link to v7 documentation, but maybe there's no good alternative (since `master` docs have a big warning label across the top). Hohum! :-)